bankdroid-legacy/src/main/java/com/liato/bankdroid/utils/ExceptionUtils.java

```diff
@@ -48,6 +39,27 @@
         return returnMe;
     }
 
+    @Nullable
+    private static <T extends Throwable> T createWrapperException(T wrap_me) {
+        Class<?> newClass = wrap_me.getClass();
+        while (newClass != null) {
+            try {
+                return (T) newClass.getConstructor(String.class)
+                        .newInstance(wrap_me.getMessage());
+            } catch (InvocationTargetException e) {
+                newClass = newClass.getSuperclass();
+            } catch (NoSuchMethodException e) {
+                newClass = newClass.getSuperclass();
+            } catch (InstantiationException e) {
+                newClass = newClass.getSuperclass();
+            } catch (IllegalAccessException e) {
+                newClass = newClass.getSuperclass();
+            }
+        }
+
+        return null;
```

Comment:
This will always return null for cases where the try block throws an exception. Is this the intention?